lib/Parse/ParseDecl.cpp

```diff
@@ -2739,6 +2739,23 @@ parseIdentifierDeclName(Parser &P, Identifier &Result, SourceLoc &Loc,
 
   P.checkForInputIncomplete();
 
+  if (P.Tok.is(tok::integer_literal) || P.Tok.is(tok::floating_literal) ||
+      (P.Tok.is(tok::unknown) && isdigit(P.Tok.getText()[0]))) {
+    // Per rdar://problem/32316666, using numbers for identifiers is a common
```

Comment:
/me doesn't like having radar references in source code. They're opaque outside of Apple. But it doesn't matter for the 4.0 branch; we can just tweak it on master separately.